src/server.js

```diff
@@ -373,6 +373,19 @@ function initializeClient(socket, client, token, lastMessage) {
 		client.unregisterPushSubscription(token);
 	});
 
+	socket.on("logs:download", (data) => {
+		const network = "irc.freenode.org"; // FIXME: Retrieve this from client one way or another
+		const logFolder = Helper.getUserLogsPath(client.name, network);
+		// TODO: When supporting entire networks, retrieve entire folder
+		const filename = path.resolve(logFolder, `${data.name}.log`);
+		const logContent = fs.readFileSync(filename, "utf8");
+
+		socket.emit("logs:download", {
```

Comment:
Sending megabytes of data over socket.io is not a great idea by the way, a better idea would be just expose logs with express and some kind of access token for that specific user.

Reply:
Yes I've considered that, but considering how sensitive this information is, I would like to avoid at all cost exposing it via Express. I can't imagine the blast radius of security fail if we have a bug with exposing this one thing.

I'm also considering always compressing this on the server before sending. Of course, there are a few channels in which that would be an issue, but I think overall this could easily work for like 95% cases. Then I think it doesn't have to be perfect at first attempt, we can refine.

Reply:
> Sending megabytes of data over socket.io is not a great idea by the way

What makes you say that, btw? I'm curious as I couldn't find evidence of that somewhere else.
There are however ways (packages and such) to send large files via sockets, or stream them, etc. Again, not thinking all the edge cases for now, just something that works.